### PR TITLE
Test fixups

### LINKED ISSUE #837: "make check" / "make test"
I see that "make test" runs lint, which uses flake8 to verify style.
Are there any functional/unit test cases that I'm overlooking?

### PR DESCRIPTION
**Pull request by mcr**
some changes to test scripts to make them easier to run, close #837

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,9 +19,10 @@ pyang/xpath_parsetab.py: pyang/xpath_parser.py
 	python -m pyang.xpath_parser
 
 test: lint
-	(cd test; $(MAKE) test)
+	which xsltproc || (echo "Please run pip install -r dev-requirements.txt";exit 1)
```
Comment (mbj4668) [inline Makefile:22]: I don't think this is a good idea; there are other tools that need to be installed, and we can assume that developers do this.  If anything, we can add documentation about what a developer needs to do.

File: Makefile [hunk 2]
```diff
@@ -19,9 +19,10 @@ pyang/xpath_parsetab.py: pyang/xpath_parser.py
...
+	(source env.sh; $(MAKE) -C test test)
```
Comment (mbj4668) [inline Makefile:23]: `$(MAKE) -C test` is fine, but we should not source env.sh here.  The tests must work w/o that.

File: pyang/plugins/sid.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 """sid plugin
 
 Plugin used to generate or update .sid files.
-Please refer to [I-D.ietf-core-sid], [I-D.ietf-core-comi], [I-D.ietf-core-yang-cbor]
+Please refer to [I-D.ietf-core-sid], [I-D.ietf-core-comi], [RFC9254]
```
Comment (mbj4668) [inline pyang/plugins/sid.py:4]: Ok, but this doesn't really belong in this PR (that supposedly fixes the tests)

File: test/Makefile [hunk 4]
```diff
@@ -12,11 +12,12 @@ export PYANG := $(PROFILE) $(W)/bin/pyang
...
 export YANG2HTML := $(PROFILE) $(W)/bin/yang2html
 else
-export PYANG := pyang
+export YANG_MODPATH:=../modules/iana:../modules/ietf
+export PYANG := $(shell pwd)/../bin/pyang
 export JSON2XML := json2xml
 export YANG2HTML := yang2html
 endif
-export YANG2DSDL := env PYANG="$(PYANG)" $(W)/bin/yang2dsdl
+#export YANG2DSDL := env PYANG="$(PYANG)" $(W)/bin/yang2dsdl
```
Comment (mbj4668) [inline test/Makefile:20]: why?

File: test/test_bad/expect/import-name.yang.out [hunk 5]
```diff
@@ -1,6 +1,5 @@
-import-name.yang:2: error: DUPLICATE_NAMESPACE
+import-name.yang:5: error: MODULE_NOT_FOUND
```
Comment (mbj4668) [inline test/test_bad/expect/import-name.yang.out:1]: This seems really weird.  This fix implies that the tests currently fail, but they don't.
This comment applies to more similar changes below.

File: test/test_sid/Makefile [hunk 6]
```diff
@@ -1,3 +1,5 @@
+PYANG?=../../bin/pyang
```
Comment (mbj4668) [inline test/test_sid/Makefile:1]: PYANG is supposed to be set when the tests are run

File: Makefile (hunk 1 above)
Comment (mcr) [reply Makefile:22]: This point is that if xsltproc isn't there, then the developer didn't read the documentation.

File: Makefile (hunk 2 above)
Comment (mcr) [reply Makefile:23]: The tests do *not* work without that.  They often either don't run at all, with PYANG='', or they run a system installed pyang, which is not the same version and lead to confusion.

File: pyang/plugins/sid.py (hunk 3 above)
Comment (mcr) [reply pyang/plugins/sid.py:4]: Yeah, okay.  It slipped in.

File: test/Makefile (hunk 4 above)
Comment (mcr) [reply test/Makefile:20]: because it overrides the one from the top, and it winds up being wrong, or redundant.

File: test/test_sid/Makefile (hunk 6 above)
Comment (mcr) [reply test/test_sid/Makefile:1]: How does it get set if you don't source env.sh?
